src/jit/lsra.cpp

```diff
@@ -4922,14 +5009,21 @@ LinearScan::allocateBusyReg(Interval *current, RefPosition *refPosition)
             // to avoid a spill - codegen can then insert the copy.
             assert(candidates == candidateBit);
             physRegNextLocation = MaxLocation;
+            farthestRefPosWeight = BB_MAX_WEIGHT;
         }
         else
         {
             physRegNextLocation = physRegRecord->getNextRefLocation();
+
+            // If refPosition requires a fixed register, we should reject all others.
+            // Otherwise, we will still evaluate all phyRegs though their next location is
+            // not better than farthestLocation found so far.
+            if (refPosition->isFixedRegRef && physRegNextLocation < farthestLocation)
+            {
+                continue;
+            }
         }
```

Comment:
If you are going to insert this code, please at least add a TODO comment that this should be using an approach similar to `tryAllocateFreeReg` where it uses a regOrder array to avoid iterating over any but the single fixed candidate in this case.